lua/nvim-autopairs/completion/compe.lua

```diff
@@ -12,11 +12,8 @@ _G.MPairs.completion_done = function()
     local _, col = utils.get_cursor()
     local prev_char, next_char = utils.text_cusor_line(line, col, 1, 1, false)
 
-    local filetype = vim.fn.expand("%:e")
-    if options.map_char[filetype] == nil then
-        filetype = "all"
-    end
-    local char = options.map_char[filetype]
+    local filetype = vim.bo.filetype
+    local char = options.map_char[filetype] or options.map_char["all"]
```

Comment:
can you add if char = '' then return . maybe the user doesn't want to insert any key on that filetype.

Reply:
The user can set the option to '', like: `map_char = { js = '' }`, and map_complete will be "disabled".

Reply:
when the char='' it still  call the feedkeys('','') .if you not return it

hmm i just found another problem you should check char inside the on_confirm_done function.
if you push this outside  the on_confirm_don mean it only executes when you call setup function.

Reply:
Okay, I'll check that